### PR TITLE
[ITBL-5597] Disable/re-register device token when email or userId changes

### PR DESCRIPTION
**Pull request by vbabenkoru**

### CODE REVIEW

File: Iterable-iOS-SDKTests/IterableAutoRegistrationTests.m [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    config.autoPushRegistration = YES;
+    [self initIterableApi];
+    [IterableAPI.sharedInstance setEmail:@"test@email.com"];
+    
+    applicationMock = OCMPartialMock([UIApplication sharedApplication]);
+    [self initIterableApi];
+    OCMVerify([applicationMock registerForRemoteNotifications]);
+}
+
+
```
Comment (tapashmajumder) [inline Iterable-iOS-SDKTests/IterableAutoRegistrationTests.m:121]: Add a test that verifies that changing from user1 => user2, disables the device for user1?

Comment (tapashmajumder) [reply Iterable-iOS-SDKTests/IterableAutoRegistrationTests.m:121]: Not needed for +1.

**Review by tapashmajumder — APPROVED** [review]
+1

**Review by davidtruong — APPROVED** [review]
LFTM

File: Iterable-iOS-SDKTests/IterableAutoRegistrationTests.m (hunk 1 above)
Comment (vbabenkoru) [reply Iterable-iOS-SDKTests/IterableAutoRegistrationTests.m:121]: It's checking whether we disable the previous device, which is the most common case (log out).

**Pull request merged**